test/tools/ossfuzz/protomutators/YulProtoMutator.h

```diff
@@ -0,0 +1,191 @@
+	static Literal* intLiteral(unsigned _value);
+
+	/// Return a variable reference
+	/// @param _seed: Pseudo-random unsigned integer used as index
+	/// of variable to be referenced
+	static VarRef* varRef(unsigned _seed);
+
+	/// Return a literal expression
+	/// @param _value: value of literal expression
+	static Expression* litExpression(unsigned _value);
+
+	/// Return a reference expression
+	/// @param _seed: Pseudo-random unsigned integer used as index
+	/// of variable to be referenced
+	static Expression* refExpression(YulRandomNumGenerator& _rand);
+
+	/// Return a load expression from location zero
+	/// @param _seed: Pseudo-random unsigned integer used to create
+	/// type of load i.e., memory, storage, or calldata.
+	static Expression* loadExpression(YulRandomNumGenerator& _rand);
+
+	static Expression* loadFromZero(YulRandomNumGenerator& _rand);
+
+	/// Configure function call from a pseudo-random seed.
+	/// @param _call: Pre-allocated FunctionCall protobuf message
+	/// @param _seed: Pseudo-random unsigned integer
+	static void configureCall(FunctionCall *_call, YulRandomNumGenerator& _rand);
+
+	/// Configure function call arguments.
+	/// @param _callType: Enum stating type of function call
+	/// i.e., no-return, single-return, multi-decl, or multi-assign.
+	/// @param _call: Pre-allocated protobuf message of FunctionCall type
+	/// @param _seed: Pseudo-random unsigned integer.
+	static void configureCallArgs(
+		FunctionCall_Returns _callType,
+		FunctionCall *_call,
+		YulRandomNumGenerator& _rand
+	);
+
+	/// Clear protobuf expression
+	/// @param _expr: Protobuf expression to be cleared
+	static void clearExpr(Expression* _expr);
+
+	/// Convert all expression-type arguments of statement
+	/// to a given type.
+	static void addArgs(
+		Statement* _stmt,
+		std::function<Expression*(YulRandomNumGenerator&)>,
+		YulRandomNumGenerator& _rand
+	);
+
+	/// Apply mutator to unset expression-type statement
+	/// arguments.
+	/// @param _stmt: Statement to be mutated
+	/// @param _seed: Pseudo-random unsigned integer
+	/// @param _mutator: Mutator function that accepts an unset expression-type
+	/// statement argument and a pseudo-random integer and applies
+	/// the mutation function to it
+	static void addArgsRec(
+		Statement* _stmt,
+		std::function<void(Expression*, YulRandomNumGenerator& _rand)> _mutator,
+		YulRandomNumGenerator& _rand
+	);
+
+	/// Add a new statement to block
+	static void addStmt(Block *_block, YulRandomNumGenerator& _rand);
+
+	/// Create binary op expression of two variable references.
+	static Expression* binopExpression(YulRandomNumGenerator& _rand);
+
+	static void unsetExprMutator(
+		Expression* _expr,
+		YulRandomNumGenerator& _rand,
+		std::function<void(Expression*, unsigned)> _mutateExprFunc
+	);
+
+	/// Check if expression is set.
+	static bool set(Expression const& _expr)
+	{
+		return _expr.expr_oneof_case() != Expression::EXPR_ONEOF_NOT_SET;
+	}
```

Comment:
The functions above are utility functions to actually perform protobuf mutations. They either modify an existing protobuf message structure or create/delete one.